jib-cli/src/main/java/com/google/cloud/tools/jib/cli/war/WarFiles.java

```diff
@@ -19,30 +19,75 @@
 import com.google.cloud.tools.jib.api.InvalidImageReferenceException;
 import com.google.cloud.tools.jib.api.Jib;
 import com.google.cloud.tools.jib.api.JibContainerBuilder;
+import com.google.cloud.tools.jib.api.buildplan.FileEntriesLayer;
 import com.google.cloud.tools.jib.cli.ArtifactProcessor;
+import com.google.cloud.tools.jib.cli.CommonCliOptions;
+import com.google.cloud.tools.jib.cli.CommonContainerConfigCliOptions;
+import com.google.cloud.tools.jib.cli.ContainerBuilders;
+import com.google.cloud.tools.jib.plugins.common.logging.ConsoleLogger;
 import com.google.common.collect.ImmutableList;
 import java.io.IOException;
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 public class WarFiles {
 
   /**
    * Generates a {@link JibContainerBuilder} from contents of a WAR file.
    *
    * @param processor artifact processor
+   * @param commonCliOptions common cli options
+   * @param commonContainerConfigCliOptions common cli options shared between jar and war command
+   * @param logger console logger
    * @return JibContainerBuilder
-   * @throws IOException if I/O error occurs when opening the war file or if temporary directory
+   * @throws IOException if I/O error occurs when opening the jar file or if temporary directory
    *     provided doesn't exist
    * @throws InvalidImageReferenceException if the base image reference is invalid
    */
-  public static JibContainerBuilder toJibContainerBuilder(ArtifactProcessor processor)
+  public static JibContainerBuilder toJibContainerBuilder(
+      ArtifactProcessor processor,
+      CommonCliOptions commonCliOptions,
+      CommonContainerConfigCliOptions commonContainerConfigCliOptions,
+      ConsoleLogger logger)
       throws IOException, InvalidImageReferenceException {
-    JibContainerBuilder containerBuilder = Jib.from("jetty");
+    JibContainerBuilder containerBuilder;
+    List<FileEntriesLayer> layers;
+    Optional<String> baseImage = commonContainerConfigCliOptions.getFrom();
+    if (baseImage.isPresent()) {
+      containerBuilder =
+          ContainerBuilders.create(
+              commonContainerConfigCliOptions.getFrom().get(),
+              Collections.emptySet(),
+              commonCliOptions,
+              logger);
+    } else {
+      containerBuilder = Jib.from("jetty");
+    }
 
-    // JVM Flags are ignored
-    List<String> entrypoint = processor.computeEntrypoint(ImmutableList.of());
+    List<String> entrypoint;
+    if (baseImage.isPresent() && !baseImage.get().startsWith("jetty")) {
+      List<String> entrypointFromCli = commonContainerConfigCliOptions.getEntrypoint();
+      entrypoint =
+          entrypointFromCli.isEmpty()
+              ? null // inherit from base image
+              : entrypointFromCli;
+    } else {
+      entrypoint = processor.computeEntrypoint(ImmutableList.of());
```

Comment:
Aha, maybe this is explicitly telling jib-core to set an empty entrypoint. I remember the jib-core API distinguishes between null (i.e., inherit) and an empty list.

Reply:
Aha that would make sense^ I just realized what the problem was. The entrypoint was correctly being set to `null` but I was setting `cmd` to an empty list instead of setting it to `null`. This is why it wasn't getting inherited from the base image.